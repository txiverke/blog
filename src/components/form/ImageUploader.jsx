// @flow

import React from 'react'

import SingleInput from './SingleInput'

class ImageUploader extends React.Component {
  state = {
    imagePreviewUrl: ''
  } 
  handleImageChange = this.handleImageChange.bind(this)

  handleImageChange(event: InputEvent) {
    event.preventDefault();
    let reader = new FileReader();
    let file = event.target.files[0];

    reader.onloadend = () => {
      this.setState({
        imagePreviewUrl: reader.result
      });
    }

    reader.readAsDataURL(file)
    this.props.handleImage(file)
  }

  render() {
    let { imagePreviewUrl } = this.state;
    let $imagePreview = null;

    if (imagePreviewUrl) {
      $imagePreview = (<img src={imagePreviewUrl} height="200" alt="Preview..." />);
    } else {
      $imagePreview = (<p className="txt">Please select an Image for Preview</p>);
    }
    
    return (
      <div className="app-grid-whole">
        <SingleInput 
          name="image"
          inputType="file"
          title="Image" 
          controlFunc={this.handleImageChange} 
          required={false}
        />
        {$imagePreview}
      </div>
    )
  }
}

export default ImageUploader