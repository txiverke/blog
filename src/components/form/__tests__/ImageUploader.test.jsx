import React from 'react'
import ImageUploader from '../ImageUploader'
import { shallow } from 'enzyme'

describe('<ImageUploader />', () => {
  it('should render itself', () => {
    const component = shallow(<ImageUploader />)
    expect(component).toMatchSnapshot()
    expect(component.text()).toEqual('<SingleInput />Please select an Image for Preview')
  })

  it('should handle events', () => {
    const component = shallow(<ImageUploader />)
    expect(component.find('SingleInput').length).toEqual(1)
  })
})