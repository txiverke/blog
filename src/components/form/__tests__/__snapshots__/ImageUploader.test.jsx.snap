// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUploader /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageUploader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SingleInput
          content=""
          controlFunc={[Function]}
          inputType="file"
          name="image"
          pattern=""
          placeholder=""
          required={false}
          size=""
          title="Image"
          wrapper="app-form-group"
/>,
        <p
          className="txt"
>
          Please select an Image for Preview
</p>,
      ],
      "className": "app-grid-whole",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": "",
          "controlFunc": [Function],
          "inputType": "file",
          "name": "image",
          "pattern": "",
          "placeholder": "",
          "required": false,
          "size": "",
          "title": "Image",
          "wrapper": "app-form-group",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select an Image for Preview",
          "className": "txt",
        },
        "ref": null,
        "rendered": "Please select an Image for Preview",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SingleInput
            content=""
            controlFunc={[Function]}
            inputType="file"
            name="image"
            pattern=""
            placeholder=""
            required={false}
            size=""
            title="Image"
            wrapper="app-form-group"
/>,
          <p
            className="txt"
>
            Please select an Image for Preview
</p>,
        ],
        "className": "app-grid-whole",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": "",
            "controlFunc": [Function],
            "inputType": "file",
            "name": "image",
            "pattern": "",
            "placeholder": "",
            "required": false,
            "size": "",
            "title": "Image",
            "wrapper": "app-form-group",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please select an Image for Preview",
            "className": "txt",
          },
          "ref": null,
          "rendered": "Please select an Image for Preview",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
